test/lib/actions/RunAction.target.test.ts

```diff
@@ -0,0 +1,169 @@
+import path = require('path');
+import {Logger} from '@salesforce/core';
+import {expect} from 'chai';
+import tildify = require('tildify');
+
+import {Results} from '../../../src/lib/output/Results';
+import {Inputs, RuleResult} from '../../../src/types';
+import {RunAction} from '../../../src/lib/actions/RunAction';
+import {InputProcessorImpl} from '../../../src/lib/InputProcessor';
+import {RuleFilterFactoryImpl} from '../../../src/lib/RuleFilterFactory';
+import {RunEngineOptionsFactory} from '../../../src/lib/EngineOptionsFactory';
+
+import {initializeTestSetup} from '../../test-related-lib/TestOverrides';
+import {FakeDisplay} from '../FakeDisplay';
+import {FakeResultsProcessorFactory, RawResultsProcessor} from './fakes';
+
+
+describe('scanner run, targeting capabilities', () => {
+	const PATH_TO_TEST_FOLDER = path.join('.', 'test', 'code-fixtures', 'apex');
+	const PATH_TO_SOME_TEST_CLASS = path.join('.', 'test', 'code-fixtures', 'apex', 'SomeTestClass.cls');
+	const PATH_TO_SOME_OTHER_TEST_CLASS = path.join('.', 'test', 'code-fixtures', 'apex', 'SomeOtherTestClass.cls');
+	const PATH_TO_SOQL_IN_LOOP = path.join('.', 'test', 'code-fixtures', 'apex', 'SoqlInLoop.cls');
+
+	let logger: Logger;
+	let display: FakeDisplay;
+	let inputProcessor: InputProcessorImpl;
+	let resultsProcessor: RawResultsProcessor;
+	let runAction: RunAction;
+
+	beforeEach(async () => {
+		initializeTestSetup();
+		logger = await Logger.child('RunAction.target.test.ts');
+		display = new FakeDisplay();
+		inputProcessor = new InputProcessorImpl('testing', display);
+		resultsProcessor = new RawResultsProcessor();
+		runAction = new RunAction(
+			logger,
+			display,
+			inputProcessor,
+			new RuleFilterFactoryImpl(),
+			new RunEngineOptionsFactory(inputProcessor),
+			new FakeResultsProcessorFactory(resultsProcessor)
+		);
+	});
+
+	describe('Test Case: Can target a single file...', () => {
+		it('...with a relative path', async () => {
+			// Prepare Input
+			const target = [PATH_TO_SOME_TEST_CLASS];
+			const inputs: Inputs = {
+				target,
+				engine: ['pmd'],
+				category: ['Performance'],
+				format: 'xml'
+			};
+
+			// Invoke tested method
+			await runAction.run(inputs);
+
+			// Assert against results
+			const results: Results = resultsProcessor.getResults();
+			assertPerformanceViolations(results, target.map(p => path.resolve(p)));
+		});
+
+		it('... with an absolute path', async () => {
+			// Prepare Input
+			const target = [path.resolve(PATH_TO_SOME_TEST_CLASS)];
+			const inputs: Inputs = {
+				target,
+				engine: ['pmd'],
+				category: ['Performance'],
+				format: 'xml'
+			};
+
+			// Invoke tested method
+			await runAction.run(inputs);
+
+			// Assert against results
+			const results: Results = resultsProcessor.getResults();
+			assertPerformanceViolations(results, target);
+		});
+
+		it('With a tilde-style path', async () => {
```

Comment:
You were using '... with' but here you have  'With'... I don't care - but I thought I'd let you know in case you did. :-)